Sources/SpeziAccountPhoneNumbers/DataEntry/OTCEntryView.swift

```diff
@@ -0,0 +1,137 @@
+    @State private var pins: [String]
+    @State private var resendTimeOut = 30
+    
+    private let timer = Timer.publish(every: 1, on: .main, in: .common).autoconnect()
+    
+    
+    var body: some View {
+        VStack {
+            HStack(spacing: 15) {
+                ForEach(0..<codeLength, id: \.self) { index in
+                    individualPin(index: index)
+                }
+            }
+            .onChange(of: pins) { _, _ in
+                updateCode()
+            }
+            .onAppear {
+                focusState = .pin(0)
+            }
+            Spacer()
+            resendSection
+            Spacer()
+        }
+            .onReceive(timer) { _ in
+                if resendTimeOut > 0 {
+                    resendTimeOut -= 1
+                }
+            }
+    }
+
+
+    private var resendSection: some View {
+        VStack {
+            Text("Didn't receive a verification code?")
+                .font(.caption)
+                .foregroundStyle(.secondary)
+            if resendTimeOut > 0 {
+                Text("Please wait for \(resendTimeOut) seconds to resend again.")
+                    .font(.caption2)
+                    .foregroundStyle(.secondary)
+            }
+            AsyncButton(action: {
+                do {
+                    guard let phoneNumber = phoneNumberViewModel.phoneNumber else {
+                        throw AnyLocalizedError(
+                            error: NSError(domain: "PhoneNumberVerification", code: 1, userInfo: nil),
+                            defaultErrorDescription: "Missing phone number"
+                        )
+                    }
+                    try await phoneVerificationProvider.startVerification(
+                        accountId: account.details?.accountId ?? "",
+                        data: StartVerificationRequest(phoneNumber: phoneNumber)
+                    )
+                    resendTimeOut = 30
+                } catch {
+                    viewState = .error(
+                        AnyLocalizedError(
+                            error: error,
+                            defaultErrorDescription: "Failed to send verification message. Please check your phone number and try again."
+                        )
+                    )
+                }
+            }) {
+                Text("Resend Verification Message")
+                    .frame(maxWidth: .infinity, minHeight: 38)
+            }
+            .disabled((phoneNumberViewModel.phoneNumber == nil) || resendTimeOut > 0)
+            .viewStateAlert(state: $viewState)
+        }
+    }
+    
+    init(codeLength: Int = 6) {
+        precondition(codeLength > 0 && codeLength <= 6, "Code length must be between one and six")
+        self.codeLength = codeLength
+        self._pins = State(initialValue: Array(repeating: "", count: codeLength))
+    }
+    
+    private func individualPin(index: Int) -> some View {
+        TextField("", text: $pins[index])
```

Comment:
To ensure you don't end up with empty strings in your localization catalog.

```suggestion
        TextField(String(), text: $pins[index])
```